osdep/macosx_application.m

```diff
@@ -106,6 +113,65 @@ - (void)dealloc
     [super dealloc];
 }
 
+- (void)setMpvHandle:(struct mpv_handle *)ctx
+{
+    _ctx = ctx;
+}
+
+#if HAVE_MACOS_TOUCHBAR
+- (NSTouchBar *)makeTouchBar
+{
+    TouchBar *tBar = [[TouchBar alloc] init];
+    [tBar setApp:self];
+    tBar.delegate = tBar;
+    tBar.customizationIdentifier = customID;
+    tBar.defaultItemIdentifiers = @[play, previousItem, nextItem, seekBar];
+    tBar.customizationAllowedItemIdentifiers = @[play, seekBar, previousItem,
+        nextItem, previousChapter, nextChapter, cycleAudio, cycleSubtitle,
+        currentPosition, timeLeft];
+    return tBar;
+}
+
+- (void)readEvents
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        while (_ctx) {
+            mpv_event *event = mpv_wait_event(_ctx, 0);
+            if (event->event_id == MPV_EVENT_NONE)
+                break;
+            [self processEvent:event];
+        }
+    });
+}
+
+-(void)processEvent:(struct mpv_event *)event
+{
+    switch (event->event_id) {
+    case MPV_EVENT_SHUTDOWN: {
+        mpv_detach_destroy(_ctx);
+        _ctx = nil;
+        break;
+    }
+    case MPV_EVENT_PROPERTY_CHANGE: {
+#if HAVE_MACOS_TOUCHBAR
```

Comment:
yeah, it is actually a copy paste fail when i moved some things around. i meant for the event loop and related to be outside of it, since it could be used in the future for other things than the Touch Bar and only touchbar related things should be inside.